### PR TITLE
FlowForma V2

### PR DESCRIPTION
**Pull request by FlowForma**
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

Please note that we are experiencing a significant delay due to a high volume of connector submissions. It may take 1-2 weeks for us to provide comments for your connector.

- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

File: certified-connectors/FlowForma V2/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# FlowForma V2
+FlowForma Business Process Management (BPM) tools for Office 365, enable users to build forms & workflow, automating processes without code.
+
```
Comment (sriyen-msft) [inline certified-connectors/FlowForma V2/README.md:3]: @FlowForma Please refer the template provided for readme.md file and add required sections.

File: certified-connectors/FlowForma V2/apiDefinition.swagger.json [hunk 2]
```diff
@@ -0,0 +1,399 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "FlowForma V2",
```
Comment (sriyen-msft) [inline certified-connectors/FlowForma V2/apiDefinition.swagger.json:4]: @FlowForma It is not allowed to create a new connector folder for the version of your existing connector. Please remove the postfix "V2" in the connector folder name and title value.

File: certified-connectors/FlowForma V2/apiDefinition.swagger.json [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+    "description": "FlowForma Business Process Management (BPM) tools for Office 365, enable users to build forms & workflow, automating processes without code.",
+    "contact": {
+      "name": "FlowForma Limited",
+      "email": "support@flowforma.com"
```
Comment (sriyen-msft) [inline certified-connectors/FlowForma V2/apiDefinition.swagger.json:8]: @FlowForma The url is missing

File: certified-connectors/FlowForma V2/apiDefinition.swagger.json [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+        "parameters": [
+          {
+            "in": "query",
+            "name": "flowId",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success"
```
Comment (sriyen-msft) [inline certified-connectors/FlowForma V2/apiDefinition.swagger.json:70]: @FlowForma Empty response schema is not allowed anymore, which means no dynamic content will be shown in the output and makers have to use parse JSON to parse the response. Please provide a valid response schema definition for all the operations in the swagger. **This applies to all the empty schemas in the swagger.**

File: certified-connectors/FlowForma V2/apiProperties.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+          "tooltip": "Provide your API Key",
+          "constraints": {
+            "tabIndex": 2,
+            "clearText": false,
+            "required": "true"
+          }
+        }
+      }
+    },
+    "iconBrandColor": "#007ee5",
```
Comment (sriyen-msft) [inline certified-connectors/FlowForma V2/apiProperties.json:18]: @FlowForma You have agreed to not to have this iconBrandColor for your connector.

image

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: certified-connectors/FlowForma V2/apiDefinition.swagger.json (hunk 2 above)
Comment (FlowForma) [reply certified-connectors/FlowForma V2/apiDefinition.swagger.json:4]: What if any effect would this have on the old connector, that was published in 2017 using a different publishing process, but has the same title?
The current connector is a new connector. That's why we called it "FlowForma V2". What would be the correct way to ensure distinguishable "V2" in the title, as seen with existing connectors?

Comment (sriyen-msft) [reply certified-connectors/FlowForma V2/apiDefinition.swagger.json:4]: @FlowForma If you look into the repo, you can see the updates applied to the existing connectors. And it is not required to add a version number to the title. Please remove it from the connector folder name and title.

https://docs.microsoft.com/en-us/connectors/custom-connectors/certification-submission#give-your-connector-a-title

File: certified-connectors/FlowForma V2/apiDefinition.swagger.json [hunk 6]
```diff
@@ -0,0 +1,585 @@
...
+  "swagger": "2.0",
+  "info": {
+    "title": "FlowForma V2",
+    "description": "FlowForma Business Process Management (BPM) tools for Office 365, enable users to build forms & workflow, automating processes without code.",
+    "contact": {
+      "name": "FlowForma Limited",
+      "url": "https://www.flowforma.com",
+      "email": "support@flowforma.com"
+    },
+    "version": "v1"
```
Comment (sriyen-msft) [inline certified-connectors/FlowForma V2/apiDefinition.swagger.json:11]: @FlowForma This should be `"V2"`.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

File: certified-connectors/FlowForma V2/apiDefinition.swagger.json (hunk 2 above)
Comment (FlowForma) [reply certified-connectors/FlowForma V2/apiDefinition.swagger.json:4]: This is a completely different connector, that's why we called it "FlowForma V2". The old "FlowForma" connector will be renamed as deprecated. These 2 connectors use completely different API architectures. so the "V2" is not a postfix, but a part of the connector's name.

**Comment by sriyen-msft** [comment]
@FlowForma Please resolve all the comments to review your PR.

File: certified-connectors/FlowForma V2/apiDefinition.swagger.json (hunk 6 above)
Comment (FlowForma) [reply certified-connectors/FlowForma V2/apiDefinition.swagger.json:11]: No, this shouldn't. the "V2" is a part of the connector's name, not the version. The API is completely new for this, so the swagger.json version should still be "v1".

**Comment by FlowForma** [comment]
@sriyen-msft I resolved all the comments

**Review by sriyen-msft — APPROVED** [review]

**Pull request merged**